File: app/src/main/java/com/arnaudpiroelle/manga/core/provider/MangaProvider.java
package com.arnaudpiroelle.manga.core.provider;

import com.arnaudpiroelle.manga.model.Chapter;
import com.arnaudpiroelle.manga.model.Manga;
import com.arnaudpiroelle.manga.model.Page;

import java.io.InputStream;
import java.util.List;

public interface MangaProvider {

    List<Manga> findMangas();

    List<Chapter> findChaptersFor(Manga manga);

    List<Page> findPagesFor(Chapter chapter);

    InputStream findPage(Page page);

    String getName();
}

File: app/src/main/kotlin/com/arnaudpiroelle/manga/core/provider/MangaProvider.kt
package com.arnaudpiroelle.manga.core.provider

import com.arnaudpiroelle.manga.model.db.Manga
import com.arnaudpiroelle.manga.model.network.Chapter
import com.arnaudpiroelle.manga.model.network.Page
import io.reactivex.Single
import okhttp3.Response

interface MangaProvider {
    fun findMangas(): Single<List<Manga>>

    fun findChapters(manga: Manga): Single<List<Chapter>>

    fun findPages(manga: Manga, chapter: Chapter): Single<List<Page>>

    fun findPage(page: Page): Single<Response>
}
